src/logcollector/read_win_event_channel.c

```diff
@@ -510,14 +351,14 @@ void send_channel_event(EVT_HANDLE evt, os_channel *channel)
             }
         }
     }
-    
+
     if (provider_name) {
         wprovider_name = convert_unix_string(provider_name);
 
-        if (wprovider_name && (msg_from_prov = get_message(evt, wprovider_name, EvtFormatMessageEvent)) == NULL) {
+        if (wprovider_name == NULL || (msg_from_prov = get_message(evt, wprovider_name, EvtFormatMessageEvent)) == NULL) {
```

Comment:
Why are we changing the logic here?

Reply:
The `get_message` function gets the message from the provider name, which means that if it does not exist, then an error should be returned. It is not necessary to check if the `wprovider_name` exists, as it is inherent to the `get_message` call, but I think it is more clear with this check.

Reply:
I think we should separate this logic to show two different error messages. It is not the same to have an empty provider name and to not be able to obtain the message. And the same error is shown in both cases (`Could not get message for ...`).

Reply:
Updated [b1b0edf](https://github.com/wazuh/wazuh/pull/3393/commits/b1b0edfd12f0b63ae59e5563dfd18be2eb049663)